Microsoft.AspNetCore.Components/Microsoft.AspNetCore.Components/EventCallback.cs

```diff
@@ -0,0 +1,154 @@
+using System;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Components
+{
+	/// <summary>
+	/// A bound event handler delegate.
+	/// </summary>
+	public readonly struct EventCallback : IEventCallback
+	{
+		/// <summary>
+		/// Gets a reference to the <see cref="T:Microsoft.AspNetCore.Components.EventCallbackFactory" />.
+		/// </summary>
+		public static readonly EventCallbackFactory Factory;
+
+		/// <summary>
+		/// Gets an empty <see cref="T:Microsoft.AspNetCore.Components.EventCallback" />.
+		/// </summary>
+		public static readonly EventCallback Empty;
+
+		internal readonly MulticastDelegate? Delegate;
+
+		internal readonly IHandleEvent? Receiver;
+
+		/// <summary>
+		/// Gets a value that indicates whether the delegate associated with this event dispatcher is non-null.
+		/// </summary>
+		public bool HasDelegate
+		{
+			get
+			{
+				throw null;
+			}
+		}
+
+		internal bool RequiresExplicitReceiver
+		{
+			get
+			{
+				throw null;
+			}
+		}
+
+		/// <summary>
+		/// Creates the new <see cref="T:Microsoft.AspNetCore.Components.EventCallback" />.
+		/// </summary>
+		/// <param name="receiver">The event receiver.</param>
+		/// <param name="delegate">The delegate to bind.</param>
+		public EventCallback(IHandleEvent? receiver, MulticastDelegate? @delegate)
+		{
+			throw null;
+		}
+
+		/// <summary>
+		/// Invokes the delegate associated with this binding and dispatches an event notification to the
+		/// appropriate component.
+		/// </summary>
+		/// <param name="arg">The argument.</param>
+		/// <returns>A <see cref="T:System.Threading.Tasks.Task" /> which completes asynchronously once event processing has completed.</returns>
+		public Task InvokeAsync(object arg)
```

Comment:
Should we make `object` nullable here for backwards-compat with the previous experience of having to call `InvokeAsync(null)` to invoke a parameter-less method?